workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs

```diff
@@ -0,0 +1,143 @@
+            }
+
+            if (ShouldConnectLocally)
+            {
+                IpAddress = GetHostIp();
+                PlayerPrefs.SetString(HostIpPlayerPrefsKey, IpAddress);
+            }
+            else
+            {
+                PlayerPrefs.DeleteKey(HostIpPlayerPrefsKey);
+            }
+
+            PlayerPrefs.Save();
+        }
+
+        /// <summary>
+        /// Extracts the Ip address that should be used to connect via the receptionist. The order is as follows:
+        /// 1. Try to extract the ip address from command line arguments passed in. This currently only works for Android.
+        /// 2. If we are running on an Android Emulator: Use the Ip address necessary to connect locally.
+        /// 3. If we are on a physical device (Android & iOS): Try to extract the value from the stored player preferences.
+        /// 4. Check if we stored anything inside the IpAddress field and use it, if we have.
+        /// 5. Return the default ReceptionistHost (localhost).
+        /// </summary>
+        /// <returns></returns>
+        protected virtual string GetHostIp()
+        {
+            var arguments = LaunchArguments.GetArguments();
+            var hostIp =
+                CommandLineUtility.GetCommandLineValue(arguments, RuntimeConfigNames.ReceptionistHost, string.Empty);
+            if (!string.IsNullOrEmpty(hostIp))
+            {
+                return hostIp;
+            }
+
+            if (Application.isMobilePlatform)
+            {
+                switch (DeviceInfo.ActiveDeviceType)
+                {
+                    case MobileDeviceType.Virtual:
+#if UNITY_ANDROID
+                        return DeviceInfo.AndroidEmulatorDefaultCallbackIp;
+#else
+                        break;
+#endif
+                    case MobileDeviceType.Physical:
+                        return PlayerPrefs.GetString(HostIpPlayerPrefsKey, IpAddress);
+                }
+            }
+
+            if (!string.IsNullOrEmpty(IpAddress))
+            {
+                return IpAddress;
+            }
+
+            return RuntimeConfigDefaults.ReceptionistHost;
+        }
+
+        protected override ConnectionService GetConnectionService()
+        {
+            return ShouldConnectLocally ? ConnectionService.Receptionist : ConnectionService.AlphaLocator;
+        }
+
+        protected override ConnectionParameters GetConnectionParameters(string workerType, ConnectionService service)
+        {
+            return new ConnectionParameters
+            {
+                WorkerType = workerType,
+                Network =
+                {
+                    ConnectionType = NetworkConnectionType.Kcp,
+                    UseExternalIp = true,
+                    Kcp = new KcpNetworkParameters
+                    {
+                        Heartbeat = new HeartbeatParameters()
+                        {
+                            IntervalMillis = 5000,
```

Comment:
These 2 values might need a comment as to why they are those values

Reply:
these values might change. Need to double-check with people, so would prefer not to do this now. The values have never changed since we introduced them btw